Moving optimizely

davidfateh opened the pull request.

jelz commented on marketplace/optimizely/src/index.css:
```diff
@@ -0,0 +1,32 @@
+
+@font-face {
+  font-family: 'Avenir Next W01';
+  src: url('https://static.contentful.com/app/avenir-regular-normal-457e94a5b6.woff2')
```
let's drop avenir and use Forma's default?

jelz commented on marketplace/optimizely/src/index.js:
```diff
@@ -0,0 +1,141 @@
+import { IncorrectContentType, isValidContentType, MissingProjectId } from './errors-messages';
+import OptimizelyClient from './optimizely-client';
+import AppPage from './AppPage';
+import '@contentful/forma-36-react-components/dist/styles.css';
```
You'll also need to import from Forma fcss to make sure headings/paragraphs have correct spacing. Shouldn't be needed soon but right now it's still required

jelz commented on marketplace/optimizely/src/index.js:
```diff
@@ -0,0 +1,141 @@
+  window.close();
+}
+
+const HOST = 'http://localhost:1234';
```
compute it with `window.location`

jelz commented on marketplace/optimizely/src/index.js:
```diff
@@ -0,0 +1,141 @@
+  render(<App sdk={sdk} />, document.getElementById('root'));
+});
+
+// if (module.hot) {
```
Just remove

jelz commented on marketplace/optimizely/src/optimizely-client.js:
```diff
@@ -0,0 +1,70 @@
+  }
+
+  makeRequest = async url => {
+    const response = await fetch(`${this.baseURL}${url}`, {
```
Let's polyfill fetch?

jelz commented on marketplace/optimizely/src/sdk.js:
```diff
@@ -0,0 +1,1543 @@
+/* eslint-disable */
```
This is a special build we don't need anymore, delete.

jelz commented on marketplace/optimizely/test/mockProps.js:
```diff
@@ -0,0 +1,1133 @@
+          type: 'Symbol',
+          required: false,
+          validations: [],
+          _defaultLocale: 'en-US',
```
Hmm, why would we need these?

jelz commented on marketplace/optimizely/package.json:
```diff
@@ -0,0 +1,46 @@
+  },
+  "dependencies": {
+    "@contentful/forma-36-fcss": "0.0.20",
+    "@contentful/forma-36-react-components": "^3.11.4",
```
Let's update forma and other deps. `npm outdated` is your friend

jelz commented:
Please rebase on top of master and fix CI.

davidfateh commented on marketplace/optimizely/extension.json:
```diff
@@ -0,0 +1,21 @@
+      {
+        "id": "contentTypeIds",
+        "type": "Symbol",
+        "name": "Comma seperated Content Types to use with Optimizely"
```
It's from the old approach, I will drop it.

davidfateh commented on marketplace/optimizely/src/AppPage/index.js:
```diff
@@ -0,0 +1,280 @@
+
+        if (index > -1) {
+          const linkValidations = validations[index];
+          const indexOfVariationContainer = linkValidations.linkContentType.indexOf(
```
The reason I use `indexOf` here is because I actually need to use it for access as well. It's not just to check for existence of the element.

davidfateh replied:
NVM 😆

davidfateh commented:
@jelz Going to be working on cleaning up the tests now. Since you last reviewed I did two things:

1. Fixed up styling around the place.
2. Cleaned up the inline oAuth flow.

If your token is expired on land in the variation container, you'll see this screen.
<img width="1680" alt="Screenshot 2019-08-02 at 08 57 38" src="https://user-images.githubusercontent.com/4086109/62350715-dbe10f80-b503-11e9-9f3b-3fa2f62ac5b0.png">

If you are working on the variation container and the token expires within 10 minutes, this banner will pop up. This happens in realtime, no need to reload the page.
<img width="1680" alt="Screenshot 2019-08-02 at 08 58 23" src="https://user-images.githubusercontent.com/4086109/62350716-dc79a600-b503-11e9-8b63-e36bcaacf624.png">

davidfateh commented on marketplace/optimizely/src/index.js:
```diff
@@ -0,0 +1,156 @@
+?client_id=15687650042
+&redirect_uri=${window.encodeURIComponent(HOST)}
+&response_type=token
+&scopes=all`;
```
No, the only option is `all`.

```
As of March 2015, the only supported value is all, meaning the generated token will have permissions that match the user role of the authorizing user.
```

davidfateh commented on marketplace/optimizely/src/index.js:
```diff
@@ -0,0 +1,156 @@
+&response_type=token
+&scopes=all`;
+
+const TOKEN_KEY = 'optToken';
```
Talked about this offline, won't do right now. Reason being: you would need to also log out of optimizely in order for the oAuth window to switch accounts.

davidfateh commented on marketplace/optimizely/src/optimizely-client.js:
```diff
@@ -0,0 +1,70 @@
+      return await response.json();
+    }
+
+    // reauthing should hopefully fix the issue
```
We actually do. When `onReauth` is called, it lets the component know to reset the `client` and subsequently ask for oAuth permissions again in the UI.

davidfateh commented on marketplace/optimizely/src/AppPage/ContentTypes.js:
```diff
@@ -0,0 +1,266 @@
+  );
+}
+
+export function isContentTypeAlreadyAdded(contentType, addedContentTypes) {
```
Turns out this wasn't being used anymore... Removed it and the `ContentTypeHelpers`
